docs/heroku.md

```diff
@@ -0,0 +1,119 @@
+
+1. Create a new app on Heroku, either via the Heroku Dashboard, or the Heroku command line.
+   For purposes of the instructions, let us suppose this is called `cs48-s20-cgaucho-lab00`
+
+2. Add a value for `SESSION_COOKIE_SECRET` to to your `.env` file.
+
+   The value can be any arbitrary string of upper and lower case letters and digits.
+   It is just a value used to encrypt your session cookies so that it's more difficult
+   for hackers to hijack your session.
+
+   Example:
+
+   ```
+   SESSION_COOKIE_SECRET=7xd6fvweSFHSS238778sf87sdfS8F8sf9ds8fDZ7sd8fdDV8ASC1232
+   ```
+
+3. Be sure you are logged into the Heroku command line via `heroku login -i`
+
+4. Run `npm install --global heroku-dotenv`
+
+   This is a utilty that provides the ability to push or pull values from `.env` to the
+   Heroku Config Vars for a Heroku app.
+
+   (On CSIL, follow instructions [here](https://github.com/sindresorhus/guides/blob/master/npm-global-without-sudo.md) before you try to use `npm install --global ...`. Note that the `.bashrc` file you need to edit is `${HOME}/.bashrc`.)
+
+5. Run this command to copy the `.env` values into Heroku Config Vars. Substitute your app
+   name in place of `cs48-s20-cgaucho-lab00`
+
+   ```
+   heroku-dotenv push --app cs48-s20-cgaucho-lab00
+   ```
+
+6. Go to the Deploy screen of the Heroku Dashboard, connect your GitHub repo
+   to the Heroku App, and then click to deploy the master branch.
+
+7. You should be up and running on Heroku. However, you still still need to modify your
+   Auth0 setup to include the new production urls.
+
+# Setting up Auth0 for Heroku (production)
+
+In order for Auth0 to recognize the app running on a production url
+running on heroku, you will need to make a small change to the Auth0
+configuration you did to get set up on `localhost`.
+
+Navigate back to the settings page of the app you created in the Auth0
+dashboard.
+
+To do this:
+
+- return to the web interface of <https://auth0.com/> and login
+- click on `Applications` in the side menu
+- select your application
+- go to the second tab for `Settings`
+
+Your production url is something of the form
+
+```
+https://cs48-s20-cgaucho-lab00.herokuapp.com
+```
+
+For every field that references `http://localhost:3000`:
+
+- Add a comma-separated entry after the existing entry referencing your new production url.
+- It is important you include **both** `localhost` **and** production urls so that both your localhost and production apps will work properly.
+
+For example, if your production url is `https://cs48-s20-cgaucho-lab00.herokuapp.com`,
+your fields should now look like this.
+
+Allowed Callback URLs:
+
+```
+http://localhost:3000/api/callback, https://cs48-s20-cgaucho-lab00.herokuapp.com/api/callback
+```
+
+Allowed Logout URLs:
+
+```
+http://localhost:3000, https://cs48-s20-cgaucho-lab00.herokuapp.com
+```
+
+Allowed Web Origins:
+
+```
+http://localhost:3000, https://cs48-s20-cgaucho-lab00.herokuapp.com
+```
```

Comment:
Are you sure this part is required? We are not using cross-origin auth in this app

Reply:
@andrewhlu

Reply:
You are correct, it isn't required. I thought we needed it since the original Next JS tutorial mentioned that we needed to include it. I'll remove it.